pkg/spanconfig/spanconfigstore/store.go

```diff
@@ -206,13 +209,26 @@ func (s *Store) ForEachOverlappingSpanConfig(
 ) error {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
-	return s.mu.spanConfigStore.forEachOverlapping(span, func(sp roachpb.Span, conf roachpb.SpanConfig) error {
+	foundNoOverlapping := true
```

Comment:
[nit] I think it might be a bit to reverse the conditional:

```
foundOverlapping := false
```

So that then below the exceptional case could be

```
if !foundOverlapping {
...
}
```